src/ForeignController.sol

```diff
@@ -330,27 +342,25 @@ contract ForeignController is ReentrancyGuard, AccessControlEnumerable {
         rateLimitedAddress(LIMIT_4626_DEPOSIT, token, amount)
         returns (uint256 shares)
     {
-        require(maxSlippages[token] != 0, "ForeignController/max-slippage-not-set");
+        require(
+            IERC4626(token).convertToAssets(1e18) <= maxExchangeRates[token],
```

Comment:
This should actually be 1 * 10 ** IERC20(token).decimals() instead of 1e18 to handle vaults with non 18 decimal precision in a saner way